packages/ember-routing/lib/location/history_location.js

```diff
@@ -150,7 +155,8 @@ export default EmberObject.extend({
    @param path {String}
   */
   pushState(path) {
-    let state = { path: path };
+    let id = this._stateCounter++;
```

Comment:
We should update this to use a UUID more details [here](https://github.com/emberjs/rfcs/pull/186#issuecomment-272524976)